src/blr/Snakefile

```diff
@@ -96,18 +95,27 @@ rule map_reads:
                 " {input.r2_fastq}",
         }
         command = commands[config["read_mapper"]].format(**locals(), **globals())
-        shell(
-            "{command} 2> >(tee {log} >&2) |"
-            "samtools sort -"
-            " -@ {threads}"
-            " -O BAM > {output.bam}"
-        )
+        if config["duplicate_marker"] == "samblaster":
+            shell(
```

Comment:
You can avoid the duplication of the entire shell command by setting a variable to `"samblaster |"` if the configured duplicate marker is samblaster and to `""` otherwise. Then just use that variable in the string that is passed to the shell command.

Reply:
That is true 👍